### PR TITLE
feat(lazer): update solana example

### PR DESCRIPTION
**Pull request by Riateche**
The example contract now calls the Pyth Lazer contract using the CPI.

### CODE REVIEW

File: lazer/solana/src/bin/client.rs [hunk 1]
```diff
@@ -49,6 +52,14 @@ fn main() -> anyhow::Result<()> {
         let signature = client.send_and_confirm_transaction(&tx)?;
         println!("OK {signature:?}");
     } else if cmd == "update" {
+        let state_data = client.get_account_data(&data_pda_key)?;
+        let state = from_bytes::<State>(&state_data);
+        println!("state: {state:?}");
```
Comment (darunrs) [inline lazer/solana/src/bin/client.rs:57]: What kinds of things are in state that we would want to print it out? I wonder if we can migrate to using tracing crate for logging to change visibility of some of these print statements.

File: lazer/solana/src/lib.rs [hunk 2]
```diff
@@ -149,14 +165,19 @@ pub fn process_update_instruction(
...
     }
-    let sysvar_account = &accounts[0];
+    let payer_account = &accounts[0];
     let data_account = &accounts[1];
-    let pyth_storage_account = &accounts[2];
+    let _pyth_program_account = &accounts[2];
+    let pyth_storage_account = &accounts[3];
+    let pyth_treasury_account = &accounts[4];
+    let system_program_account = &accounts[5];
+    let instructions_sysvar_account = &accounts[6];
```
Comment (darunrs) [inline lazer/solana/src/lib.rs:177]: What do all these programs do? Are they actually different accounts entirely?

**Review by darunrs — COMMENTED** [review]
Truth be told, I'm not really familiar with what is going on here. Mainly, I see we working with new accounts. Is this related to the addition of fees to verification?

**Review by ali-bahjati — APPROVED** [review]
Nice!

File: lazer/solana/src/lib.rs (hunk 2 above)
Comment (Riateche) [reply lazer/solana/src/lib.rs:177]: Program accounts are required to allow the contract to call them. This contract calls the pyth contract (`_pyth_program_account `) which calls the system program (`system_program_account`) to make the fee transfer. `pyth_storage_account` and `pyth_treasury_account` are accounts that have to be accessed by the pyth contract. `instructions_sysvar_account` is a special account that allows programs to inspect other instructions within the current transaction, which is needed to validate the signature.

**Pull request merged**